src/NuGet.Core/NuGet.Packaging/Signing/TrustedSigners/TrustedSignersProvider.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using NuGet.Common;
+using NuGet.Configuration;
+
+namespace NuGet.Packaging.Signing
+{
+    public static class TrustedSignersProvider
+    {
+        public static IReadOnlyList<VerificationAllowListEntry> GetAllowListEntries(ISettings settings, ILogger logger)
```

Comment:
None of the `ISettings` operations are async. We could discuss if we want to add async support for `ISettings`... Since it is a read operation is working in the computed settings, therefore there is no need for it to be async.